sdk/tables/azure-data-tables/tests/test_table_entity_cosmos.py

```diff
@@ -1971,11 +1979,11 @@ def test_entity_with_edmtypes(self, tables_cosmos_account_name, tables_primary_c
             self.table.upsert_entity(entity)
             result = self.table.get_entity(entity["PartitionKey"], entity["RowKey"])
             assert result["bool"] == False
-            assert result["text"] == "42"
+            assert result["text"] == 42
             assert result["number"] == 23
             assert result["bigNumber"][0] == 64
-            assert result["bytes"] == b"test"
-            assert result["amount"] == 0.0
+            assert _encode_base64(result["bytes"]) == "test"
+            assert result["amount"] == 0
```

Comment:
hmm this test shouldn't have changed....

Reply:
The `new` encoded entity of this test entity is:
```json
{
  "PartitionKey": "pkbc132fa",
  "RowKey": "rkbc132fa",
  "bool@odata.type": "Edm.Boolean",
  "bool": "false",
  "text@odata.type": "Edm.String",
  "text": 42,
  "number@odata.type": "Edm.Int32",
  "number": "23",
  "bigNumber@odata.type": "Edm.Int64",
  "bigNumber": "64",
  "bytes@odata.type": "Edm.Binary",
  "bytes": "test",
  "amount@odata.type": "Edm.Double",
  "amount": "0",
  "since@odata.type": "Edm.DateTime",
  "since": "2008-07-10T00:00:00",
  "guid@odata.type": "Edm.Guid",
  "guid": "c8924069-90ee-4726-83b5-45ff92496126"
}
```
while old encoded result is:
```json
{
  "PartitionKey": "pkbc132fa",
  "PartitionKey@odata.type": "Edm.String",
  "RowKey": "rkbc132fa",
  "RowKey@odata.type": "Edm.String",
  "bool": "false",
  "bool@odata.type": "Edm.Boolean",
  "text": "42",
  "text@odata.type": "Edm.String",
  "number": 23,
  "number@odata.type": "Edm.Int32",
  "bigNumber": "64",
  "bigNumber@odata.type": "Edm.Int64",
  "bytes": "dGVzdA==",
  "bytes@odata.type": "Edm.Binary",
  "amount": "0",
  "amount@odata.type": "Edm.Double",
  "since": "2008-07-10T00:00:00",
  "since@odata.type": "Edm.DateTime",
  "guid": "c8924069-90ee-4726-83b5-45ff92496126",
  "guid@odata.type": "Edm.Guid"
}
```

Reply:
These changes are because the new encoder for tuple case is not backward compatible, we probably should convert the value based on the edm type in tuple.

Reply:
We've resolved the tuple backward compatible issue in the latest commit.